stew/base36.nim

```diff
@@ -0,0 +1,238 @@
+## Copyright (c) 2025 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+## This module implements Base36 encoding and decoding procedures.
+## This module supports two variants of Base36 encoding (Bitcoin and Flickr).
+
+type
+  Base36Status* {.pure.} = enum
+    Error,
+    Success,
+    Incorrect,
+    Overrun
+
+  Base36Alphabet* = object
+    decode*: array[128, int8]
+    encode*: array[36, uint8]
+
+  Base36* = object
+  Base36Uc* = object
+  Base36Lc* = object
+
+  Base36Types* = Base36 | Base36Uc | Base36Lc
+
+  Base36Error* = object of CatchableError
+    ## Base36 specific exception type
+
+func newAlphabet36(s: string): Base36Alphabet =
+  doAssert(len(s) == 36)
+  var alphabet: Base36Alphabet
+  for i in 0..<len(s):
+    alphabet.encode[i] = cast[uint8](s[i])
+  for i in 0..<len(alphabet.decode):
+    alphabet.decode[i] = -1
+  for i in 0..<len(alphabet.encode):
+    alphabet.decode[int(alphabet.encode[i])] = int8(i)
+  return alphabet
+
+const
+  B36UcAlphabet* = newAlphabet36("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ")
+  B36LcAlphabet* = newAlphabet36("0123456789abcdefghijklmnopqrstuvwxyz")
+
+func encodedLength(btype: typedesc[Base36Types], length: int): int =
```

Comment:
Neither `encodedLength` nor `decodedLength` appear to be used?

Reply:
They seem to be required (base64, base58, base32 have the same functions) I'm marking them as public

Reply:
None of those modules is a good example to copy.